treeherder/push_health/usage.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from treeherder.config import settings
+from treeherder.model.models import Push
+from treeherder.utils.http import make_request
+from treeherder.webapp.api.serializers import PushSerializer
+
+logger = logging.getLogger(__name__)
+
+
+def get_peak(facet):
+    peak = 0
+    date = 0
+    for item in facet['timeSeries']:
+        max = item['results'][-1]['max']
+        if item['inspectedCount'] > 0 and max > peak:
+            peak = max
+            date = item['endTimeSeconds']
+
+    return {'needInvestigation': peak, 'time': date}
+
+
+def get_latest(facet):
+    for item in reversed(facet['timeSeries']):
+        if item['inspectedCount'] > 0:
+            latest = item['results'][-1]
+            return {
+                'needInvestigation': latest['max'],
+                'time': item['endTimeSeconds']
+            }
+
+
+def get_usage(api_key=None):
+    nr_api_key = api_key or settings.NEW_RELIC_INSIGHTS_API_KEY
+    nrql = "SELECT%20max(needInvestigation)%20FROM%20push_health_need_investigation%20FACET%20revision%20SINCE%201%20DAY%20AGO%20TIMESERIES%20where%20repo%3D'{}'%20AND%20appName%3D'{}'".format(
+        'try', 'treeherder-prod')
+    new_relic_url = '{}?nrql={}'.format(settings.NEW_RELIC_INSIGHTS_API_URL, nrql)
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'X-Query-Key': nr_api_key,
+    }
+    if not nr_api_key:
+        exception = EnvironmentError('NEW_RELIC_INSIGHTS_API_KEY not set on server.')
```

Comment:
Should this kind of issue be caught in `treeherder.config.settings`?

Reply:
I'm not saying it should but want to make sure that it was intentional.

Reply:
Yeah, interesting question.  I think I won't have it bomb in `config.settings` in case someone wants to run Treeherder locally and doesn't care about Push Health.  They won't be required to set that value.  I'll just log the error there.

But I'll add a bomb out in `pre_deploy` so we don't deploy to Heroku without it.  How does that sound to you?

Reply:
I know exactly what you mean. Whatever you're confortable with.
Eventually I have plans that Travis will test "./manage.py check" for all 4 combinations: production, review app, within Docker and outside Docker.